src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/PensionAnnexDateStampService.java

```diff
@@ -0,0 +1,91 @@
+                                                     String authToken,
+                                                     LocalDate approvalDate,
+                                                     String caseId) throws Exception {
+        log.info("Adding date stamp to Pension Sharing Annex : {}", document);
+        Optional<LocalDate> optionalApprovalDate = Optional.ofNullable(approvalDate);
+        if (optionalApprovalDate.isPresent()) {
+            byte[] docInBytes = emDownloadService.download(document.getDocumentBinaryUrl(), authToken);
+            byte[] approvedDoc = appendApprovedDateToDocument(docInBytes, approvalDate);
+            MultipartFile multipartFile =
+                FinremMultipartFile.builder().name(document.getDocumentFilename()).content(approvedDoc)
+                    .contentType(APPLICATION_PDF_VALUE).build();
+            List<FileUploadResponse> uploadResponse =
+                emUploadService.upload(Collections.singletonList(multipartFile), caseId, authToken);
+            FileUploadResponse fileSaved = Optional.of(uploadResponse.get(0))
+                .filter(response -> response.getStatus() == HttpStatus.OK)
+                .orElseThrow(() -> new DocumentStorageException("Failed to store document"));
+            Document dateStampedDocument = CONVERTER.apply(fileSaved);
+            return genericDocumentService.toCaseDocument(dateStampedDocument);
+        } else {
+            throw new StampDocumentException("Missing or Invalid Approved Date of Order.");
+        }
+    }
+
+    private byte[] appendApprovedDateToDocument(byte[] inputDocInBytes, LocalDate approvalDate) throws StampDocumentException, IOException {
+        PDDocument doc = Loader.loadPDF(inputDocInBytes);
+        doc.setAllSecurityToBeRemoved(true);
+        Optional<PDAcroForm> acroForm = Optional.ofNullable(doc.getDocumentCatalog().getAcroForm());
+        if (acroForm.isPresent() && acroForm.get().getField(FORM_P1_DATE_OF_ORDER_TEXTBOX_NAME) != null
+            && (acroForm.get().getField(FORM_P1_DATE_OF_ORDER_TEXTBOX_NAME) instanceof PDTextField)) {
+            PDField field = acroForm.get().getField(FORM_P1_DATE_OF_ORDER_TEXTBOX_NAME);
+            PDTextField textBox = (PDTextField) field;
+            if (textBox.getDefaultAppearance() == null && textBox.getDefaultAppearance().isEmpty()) {
+                textBox.setDefaultAppearance(DEFAULT_PDTYPE_PREFIX + DEFAULT_PDTYPE_FONT_HELV + DEFAULT_PDTYPE_POSTFIX);
+            }
+            textBox.setValue(approvalDate.format(DateTimeFormatter.ofPattern(DATE_STAMP_PATTERN).withLocale(Locale.UK)));
+            ByteArrayOutputStream outputBytes = new ByteArrayOutputStream();
+            doc.save(outputBytes);
+            doc.close();
+            return outputBytes.toByteArray();
+        } else {
+            throw new StampDocumentException("Pension Order document PDF is flattened / not editable.");
+        }
+    }
+}
```

Comment:
looks like textBox.getDefaultAppearance() returns a string.  So you could try isBlank rather than the null and empty check.